rust-update

```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
+
+if [ "$1" = '-h' ] ; then
+	echo 'Updates rustc & clippy'
+	echo 'It first checks if clippy would compile at currentl nightly and if so, it updates.'
+	echo 'Options:'
+	echo '-h: This help message'
+	echo '-f: Skips the check and just updates'
+	exit
+fi
+
+set -ex
+
+renice -n 10 -p $$
+
+export CARGO_INCREMENTAL=0
+export RUSTFLAGS='-C target-cpu=native'
+
+try_out() {
+	export RUSTUP_HOME=$HOME/.rustup-attempt
+	test -d $RUSTUP_HOME || (rustup toolchain add nightly && rustup default nightly)
+	rustup update
```

Comment:
Should we explicitly `rustup update nightly`, as that is what clippy uses, and other toolchains could perhaps sneak in if someone does a rustup update in parallel in the same shell (yes, that's quite contrived, but better safe than sorry).

Reply:
I don't see what you mean „in the same shell“, as the script runs in its own instance of sh, so it's impossible to run any „user“ commands in it.

However, I think I guess the general direction of where you lead with the question. `rustup update` actually updates all the toolchains (and components and targets) if you don't specify one, which is *probably* what the user wants anyway (it is what *I* want for sure), no matter which one is your default.

On the other hand, it is not guaranteed that user's default toolchain is nightly, which this script assumed. I didn't modify the `rustup update` call, but I added `+nightly` in the cargo command, to make sure clippy is compiled with that.

Does that address your concern, or did I misunderstood the problem?

Reply:
No, that's fine.